tutorials/data-lake-sof-multiple-directories/data-lake-sof-multiple-directories.md

```diff
@@ -0,0 +1,238 @@
+auto_validation: true
+time: 45
+tags: [ tutorial>intermediate, software-product>sap-hana]
+primary_tag: software-product-function>sap-hana-cloud\,-data-lake
+author_name: Ronit Nagwekar
+author_profile: https://github.com/rnagweka
+parser: v2
+---
+
+## Prerequisites
+-	Have access to a licensed managed or standalone SAP HANA data lake.
+-	Installation of the HDLFSCLI. See tutorial Getting Started with Data Lake Files HDLFSCLI | Tutorials for SAP Developers.
+-	Load some structured data files inside of a SAP HANA data lake File Container.
+-	Already set up HDLFS Connection in Database Explorer - Setting Up HDLFS Connection In Database Explorer.
+
+
+## You will learn
+- How to define the database objects required to query structured data files.
+- How to add a data source for a folder and subfolder directory and query the output of that file.
+
+
+## Intro
+
+Querying structured data files (CSV, ORC, Parquet) in a HANA Data Lake file container can be done using SQL on Files. Below you will find all of the steps required to start using SQL on Files.
+If you have not yet provisioned an SAP HANA data lake, here is a great tutorial on how to do so!
+Please go through the entire tutorial on **Use SQL on Files to query data from a single file/directory**
+
+We will be using the Orders table as a reference.
+
+---
+
+### STEP 1: Create a SQL on Files remote table in Files service
+
+Note that the table definition will need to match the expected schema/data types of the columns in the structured file you are trying to query. Login to the SQL console as User1 or HDLADMIN and execute all the queries associated with this tutorial. I will be executing the queries through HDLADMIN.
+
+Firstly, we create a SCHEMA in files service.
+
+```SQL
+CREATE SCHEMA HDLADMIN_TPCH_SQLONFILES IN FILES_SERVICE;
+```
+
+Let us CREATE THE TABLE now
+
+```SQL
+CREATE TABLE HDLADMIN_TPCH_SQLONFILES.HDLRE_ORDERS_SQLONFILES
+(
+ORDERYEAR		varchar(4),
+ORDERMONTH	varchar(2),
+O_ORDERKEY           bigint,
+O_CUSTKEY              integer,
+O_ORDERSTATUS     varchar(2),
+O_TOTALPRICE         decimal(15,2),
+O_ORDERDATE          date,
+O_ORDERPRIORITY   varchar(15),
+O_CLERK                    varchar(15),
+O_SHIPPRIORITY       integer,
+O_COMMENT             varchar(79)   
+) IN FILES_SERVICE;
+```
+<!-- border --> ![DBX Screenshot](image-1.png)
+
+For information about the parameter definitions and supported data types, see  [CREATE (Remote) TABLE Statement for Data Lake Relational Engine (HANA DB-Managed)](help.sap.com/docs/SAP_HANA_DATA_LAKE/a898e08b84f21015969fa437e89860c8/24e694b566814ad285cb32fe3e5d3928.html?state=DRAFT&version=2022_1_QRC)
+
+
+Next, we create a virtual table. Notice in the SQL below where the remote servers name goes and where the reference to the table in the Files Service goes. Over here, we will be creating an ORDERS VIRTUAL TABLE  in HDLRE that points to the ORDERS table that we just created in SQL On Files service.
```

Comment:
Please don't use the "royal we" when addressing the user. Use "you" instead.
Please check the [style guide](https://jam4.sapjam.com/wiki/show/MozDWaPczznL8NKeGEk0y9#mcetoc_1dtvf24ta2) for details.